applications/nrf5340_audio/src/audio/audio_datapath.h

```diff
@@ -17,9 +17,8 @@
 /* Presentation delay defines in microseconds */
 /* Allow some buffer time to allow for HCI Transport etc */
 #define PRES_DLY_BUFFER_US 2500
-#define MAX_PRES_DLY_US 40000
-#define DEFAULT_PRES_DLY_US 10000
-#define MIN_PRES_DLY_US (DEC_TIME_US + PRES_DLY_BUFFER_US)
+#define ABSOLUTE_MAX_PRES_DLY_US 40000
```

Comment:
I used absolute as this is the limit determined by the hardware and buffering and will be set at manufacture. Will revert to min and max.
The max pres delay is used to calculate the buffer size which is the reverse of the min which is set by the decoder latency (audio processing) and pres delay buffering. I dont actually see where this pres delay buffering is used, other than here, so is there a reason for it to be 2500us?
So do I remove these from the file and add to a kconig and determine the pres delay buffer size from this, taking the decoder delay into account. It would probably be wise to add the dec time and any other audio processing delay to a kconfig, as this will very between codec suppliers . It is currently determined by software or hardware decode type in a kconfig. Comments?

Reply:
You are right, the 2500 is just a value that was selected as a finger in the air buffer time to allow hci transfer, fifo, pcsm etc. The dec time is also set based on measurements of the current SW codec and could change with different sample rates, bit rates etc. Don't know what the best method to set these values are

Reply:
Usually the codec is profiled over a set of test vectors and sample audio. This gives a table of cycles/MIPS/MHz and then the user has the data to decide if the codec can operate within resource range. This dose assume that the codec is the only thing running! Did we get that in this case for the codec?
The remaining audio path will be similarly calculated and then a fudge factor added. This total would then give min/max range for the system, roughly. It is then rigorously tested.
So in my experience we pass this to the user/customer as they know their system, thus we expose this in the Kconfig as CONFIG_AUDIO_MIN_PRES_DLY_US and CONFIG_AUDIO_MAX_PRES_DLY_US with associated caveats.